source/h1_connection.c

```diff
@@ -1339,16 +1339,17 @@ static int s_handler_process_read_message(
                 aws_last_error(),
                 aws_error_name(aws_last_error()));
 
-            goto error;
+            goto shutdown;
         }
     }
 
     aws_mem_release(message->allocator, message);
     return AWS_OP_SUCCESS;
-error:
+
+shutdown:
     aws_mem_release(message->allocator, message);
     s_shutdown_connection(connection, aws_last_error());
-    return AWS_OP_ERR;
+    return AWS_OP_SUCCESS;
```

Comment:
So I see raising errors, and then returning success anyway... how does that work out?

Reply:
Per Mike, errors are passed to shutdown, but we return sucess to indicate we consumed the message.  There's some talk of figuring out how to communicate/enforce the io message contract better (error = sender frees, success = recipient frees)

Reply:
Perhaps add some comments to that effect, so the next person isn't confused?